include/posix/pthread.h

```diff
@@ -7,39 +7,42 @@
 #ifndef __PTHREAD_H__
 #define __PTHREAD_H__
 
-#ifdef CONFIG_NEWLIB_LIBC
+#include <kernel.h>
 #include <time.h>
-#else
-/* This should probably live somewhere else but Zephyr doesn't
- * currently have a stdc layer to provide it
+#include "sys/types.h"
+#include "posix_sched.h"
+#include "unistd.h"
+
+struct posix_thread {
+	struct k_thread thread;
+	void *retval;
+};
+
+/* Pthread cancellation */
+#define PTHREAD_CANCEL_POS     0
+/* Non Essential Thread */
+#define PTHREAD_CANCELABLE     (0 << PTHREAD_CANCEL_POS)
+/* Essential Thread */
+#define PTHREAD_NON_CANCELABLE (1 << PTHREAD_CANCEL_POS)
```

Comment:
If I search for `PTHREAD_NON_CANCELABLE`, I only find a reference to this pull request. If this is an implementation detail, please move its definition to the `.c` file.